File: solution/2090. K Radius Subarray Averages/2090.cpp
class Solution {
public:
    vector<int> getAverages(vector<int>& nums, int k) {
        const int n = nums.size();
        vector<int> res(n, -1);
        int len = k + k;
        long sum = 0;
        for (int i = 0; i < n; ++i) {
            sum += nums[i];
            if (i >= len) {
                res[i - k] = sum / (len + 1);
                sum -= nums[i - len];
            }
        }
        return res;
    }
};

File: solution/2090. K Radius Subarray Averages/2090.py
class Solution:
    def getAverages(self, nums: List[int], k: int) -> List[int]:
        res = [-1] * len(nums)
        l = k + k
        s = 0
        for i, x in enumerate(nums):
            s += x
            if i >= l:
                res[i - k] = s // (l + 1)
                s -= nums[i - l]
        return res
